src/protocol/mod.rs

```diff
@@ -193,18 +193,29 @@ where
     }
 }
 
-/// A trait for attempting to create a `Message` from `self`. This is required to be implemented
-/// on a broker's `Delivery` type.
+/// A trait for attempting to create a `Message` from `self`. This will be implemented
+/// by types that can act like message "factories", like for instance the `Signature` type.
 pub trait TryCreateMessage {
     fn try_create_message(&self) -> Result<Message, ProtocolError>;
 }
 
-impl TryCreateMessage for Message {
+impl<T> TryCreateMessage for Signature<T>
+where
+    T: Task + Clone,
+{
+    /// Creating a message from a signature without consuming the signature requires cloning it.
+    /// For one-shot conversions, directly use `Message::try_from` instead.
     fn try_create_message(&self) -> Result<Message, ProtocolError> {
-        Ok(self.clone())
+        Message::try_from(self.clone())
     }
 }
 
+/// A trait for attempting to deserialize a `Message` from `self`. This is required to be implemented
+/// on a broker's `Delivery` type.
+pub trait TryDeserializeMessage {
```

Comment:
This trait was previously called `TryCreateMessage`. I have changed its name trying to highlight this difference:
- the `TryCreateMessage` is a message factory which creates new messages to be sent
- the `TryDeserializeMessage` is used to deserialize a received message in order to execute it